server/datastore/mysql/android_test.go

```diff
@@ -357,3 +359,99 @@ func testAndroidHostStorageData(t *testing.T, ds *Datastore) {
 	assert.Equal(t, 64.0, updatedFullHost.GigsDiskSpaceAvailable, "Updated available disk space should be saved in host_disks")
 	assert.Equal(t, 25.0, updatedFullHost.PercentDiskSpaceAvailable, "Updated disk space percentage should be saved in host_disks")
 }
+
+func testNewAndroidHostWithIdP(t *testing.T, ds *Datastore) {
+	test.AddBuiltinLabels(t, ds)
+
+	// create IdP account... InsertMDMIdPAccount generates its own UUID
+	idpAccount := &fleet.MDMIdPAccount{
+		Username: "john.doe",
+		Fullname: "John Doe",
+		Email:    "john.doe@example.com",
+	}
+	err := ds.InsertMDMIdPAccount(context.Background(), idpAccount)
+	require.NoError(t, err)
+
+	// get the actual UUID that was generated
+	insertedAccount, err := ds.GetMDMIdPAccountByEmail(context.Background(), "john.doe@example.com")
+	require.NoError(t, err)
+	require.NotNil(t, insertedAccount)
+	idpAccount.UUID = insertedAccount.UUID
+
+	// create Android host
+	const enterpriseSpecificID = "enterprise_with_idp"
+	host := &fleet.AndroidHost{
```

Comment:
I believe there's a `createAndroidHost` helper that you may be able to use here.

Reply:
I have it in my branch 👍